### PR TITLE
Add toast with option to open post after publishing in web UI

### PR DESCRIPTION
**Pull request by Gargron**
Provide feedback on successful publishing with a way to open the new post. Redesign the toast component.

Screenshot 2023-06-24 at 17-00-18 Foo (@admin@localhost)

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
The code looks fine, and I thought I liked the idea, which can help with at least the onboarding page, but upon testing:
- it does feel distracting, especially on mobile where this takes a significant portion of the screen
- on a typical 16:9 screen on the default single-column view, the toast will appear pretty far from anything, thus might not get noticed and/or might need significant eye movement

image

I'm also not sure the visual redesign is necessary and I'm not entirely convinced with the way it looks with regards to transparency and blur:
image

File: app/javascript/styles/mastodon/components.scss [hunk 1]
```diff
@@ -9010,3 +9010,61 @@ noscript {
...
+  box-shadow: 0 10px 15px -3px rgba($base-shadow-color, 0.25),
+    0 4px 6px -4px rgba($base-shadow-color, 0.25);
+  cursor: default;
+  transition: 0.5s cubic-bezier(0.89, 0.01, 0.5, 1.1);
+  transform: translateZ(0);
+  font-size: 15px;
+  line-height: 21px;
+
+  &.notification-bar-active {
+    left: 1rem;
```
Comment (ClearlyClaire) [inline app/javascript/styles/mastodon/components.scss:9044]: This isn't consistent with `inset-inline-start: -100%` and will not override it.

File: app/javascript/styles/mastodon/components.scss [hunk 2]
```diff
@@ -9010,3 +9010,61 @@ noscript {
...
+  left: 0;
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+}
+
+.notification-bar {
+  flex: 0 0 auto;
+  position: relative;
+  inset-inline-start: -100%;
```
Comment (ClearlyClaire) [inline app/javascript/styles/mastodon/components.scss:9026]: This is not consistent with `left: 1rem` at https://github.com/mastodon/mastodon/pull/25564/files#diff-edd345ee45838ecdc48d8b64ba5f0d4475300e2cc36e696a937fe012a6e025caR9044 and will not get overridden on RTL layouts. Furthermore, this is not consistent with `left: 0` at https://github.com/mastodon/mastodon/pull/25564/files#diff-edd345ee45838ecdc48d8b64ba5f0d4475300e2cc36e696a937fe012a6e025caR9017 and will not be positioned properly on RTL layouts.

File: app/javascript/styles/mastodon/components.scss [hunk 3]
```diff
@@ -9010,3 +9010,61 @@ noscript {
     }
   }
 }
+
+.notification-list {
+  position: fixed;
+  bottom: 2rem;
+  left: 0;
```
Comment (ClearlyClaire) [inline app/javascript/styles/mastodon/components.scss:9017]: This is not consistent with `inset-inline-start` a few lines later.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
There are inconsistent uses of `inset-inline-start` and `left` that will cause issues on RTL layouts.

**Comment by Gargron** [comment]
Made the background of the toast darker.

**Pull request merged**